components/Feature/VulnerabilitiesGrid/index.js

```diff
@@ -10,16 +10,33 @@ import groups from './grid.json';
 const VulnerabilitiesGrid = ({ onUpdate }) => {
   const [grid, setGrid] = useState({ assets: {}, vulnerabilities: {} });
 
+  const addItem = (obj, key, value) => {
+    obj[key] = value;
+    return obj;
+  };
+
+  const removeItem = (obj, key) => {
+    return Object.fromEntries(Object.entries(obj).filter(([k]) => k != key));
+  };
```

Comment:
If you modelled this as a `Map` you could do this slightly quicker, doing `fromEntries` on an `entries` that is filtered will iterate through all the keys maybe up to 3 times.

Probably doesn't really matter given the size of the objects we will be working with, but throwing it out there if you're interested.

Here's a slightly simplified example...

```
  // assuming we can turn assets into a map when we initialise the grid state
  const assets = new Map(Object.entries(someObjectOfInitialAssets));

  // add an item
  assets.set(key, value);

  // remove an item
  assets.delete(key);
```

Reply:
Nice, I might refactor this tonight